test/functional/test_framework/test_framework.py

```diff
@@ -108,6 +108,22 @@ def main(self):
             os.makedirs(self.options.tmpdir, exist_ok=False)
         else:
             self.options.tmpdir = tempfile.mkdtemp(prefix="test")
+
+        exit_code = TEST_EXIT_PASSED
+        for variant in self.variants:
+            self.variant = variant
+            self.tmpdir = self.options.tmpdir
+            if variant:
+                self.tmpdir = os.path.join(self.options.tmpdir, variant)
+                os.mkdir(self.tmpdir)
+            success = self.run_variant()
```

Comment:
I'd prefer if this was renamed to `status`.

Also, we don't support different statūs for variants, since the test runner has to print one status per test script. Thus, I'd prefer if the result of `run_variant` was appended to the list `status` and then (outside the for loop):

* All statūs the same => `sys.exit(TEST_EXIT_${STATUS})`
* Different statūs, probably best to exit with failure.

Reply:
Will rename. The behavior you're describing is what I was intending and what should happen after fixing the bug below (no need for an array/list just to return one status).